### PR TITLE
Add new method Wikimate::undelete()

### PR DESCRIPTION
**Pull request by Xymph**
New support for API call, with version-dependent `undeletetalk` parameter.
This clears my backlog of prepared code :) and I think it's a good time to wrap up v1.1.0.

### CODE REVIEW

File: Wikimate.php [hunk 1]
```diff
@@ -707,6 +708,44 @@ public function delete($array)
         return $this->request($array, $headers, true);
     }
 
+    /**
+     * Performs an undelete query to the wiki API.
+     *
+     * @param   string         $name     The original name of the wiki article or file
+     * @param   string         $reason   Reason for the restore
+     * @param   boolean        $talktoo  True to restore talk page, if any
```
Comment (waldyrious) [inline Wikimate.php:716]: Just a tiny nitpick :smile:

```suggestion
     * @param   boolean        $talktoo  True to restore talk page, if any;
```

File: Wikimate.php [hunk 2]
```diff
@@ -707,6 +708,44 @@ public function delete($array)
         return $this->request($array, $headers, true);
     }
 
+    /**
+     * Performs an undelete query to the wiki API.
+     *
+     * @param   string         $name     The original name of the wiki article or file
```
[the remark this answers is not in the document]
Comment (Xymph) [reply Wikimate.php]: Both terms are used [interchangeably by MediaWiki](https://www.mediawiki.org/wiki/Help:Starting_a_new_page) :
"the edit page for the new article", "indicating that the article now exists", "The URL to an article of the wiki", etc. 'name' is also used interchangeably with 'title' for articles/pages (same URL). But, a file has a name not a title, so the existing comment line covers both cases in a condensed way.

Wikipedia also [calls them articles](https://en.wikipedia.org/wiki/Help:Your_first_article). On the other hand, Wikimate can also handle stuff outside main space like categories, and those are pages indeed. So this is a bit of a conundrum.

Also, in Wikimate.php comments, we already use "wiki article" a few times (e.g. `Wikimate::getPage()` and `WikiPage::__construct()` which have a $title parameter...). And even the [slightly awkward "wiki article page"](https://github.com/Xymph/Wikimate/blob/153-add-undelete/Wikimate.php#L875) which goes back to the class being named WikiPage and initially being aimed at (main space) articles.

So long story short, I changed to "The original name of the wiki page or file".

[the remark this answers is not in the document]
Comment (waldyrious) [reply Wikimate.php]: Yeah, I noticed the existing usages of "article" in Wikimate, but didn't want to bring a full reworking of that term into this PR; I just tried not to compound the issue.

> Wikipedia also [calls them articles](https://en.wikipedia.org/wiki/Help:Your_first_article).

Yes, but Wikipedia is an encyclopedia. For different wikis (or different namespaces, as you point out) each page may be something different semantically — an "entry", a "post", a "profile", etc.; but regardless, they are always an individual web page. That's why I consider "page" to be a better term to use generically, especially in the context of interacting with the API.

As for name vs title, fair enough — I always considered the human-readable rendering of a page URL as its title, as opposed to the name, which can be thought of a more semantically meaningful descriptor (e.g. to me, "File:Foobar.jpg" is the page title, and "Foobar" is the page name). But this is not as clear cut, and it's a minor distinction anyway. I'm happy with the wording you chose, unless you want to flesh out the subject some more (which I'm happy to do as well).

[the remark this answers is not in the document]
Comment (Xymph) [reply Wikimate.php]: I think it suffices for this PR, but if you want to propose changes throughout Wikimate (preferably minimalistic though), then feel free to use mail or open a PR.

[the remark this answers is not in the document]
Comment (waldyrious) [reply Wikimate.php]: Maybe sometime in the future. We've got more important things to attend to :)

**Review by waldyrious — APPROVED** [review]
LGTM!

**Comment by Xymph** [comment]
Alright, so agree to wrap up v1.1.0?
This needs to include a link to #153 in Changelog, since I forgot that this time.

**Comment by waldyrious** [comment]
> Alright, so agree to wrap up v1.1.0?

Yep! Let's go!

> This needs to include a link to https://github.com/hamstar/Wikimate/pull/153 in Changelog, since I forgot that this time.

Ah, interesting. We might want to add some sort of markdown validation in the changelog checks, so that we confirm not just the presence but also the syntactic validity. I believe [markdownlint](https://github.com/DavidAnson/markdownlint) is able to do this. (We'd have to disable the rule about duplicate headings though, by adding `<!-- markdownlint-disable no-duplicate-heading -->` to the top of the changelog file.)

**Comment by waldyrious** [comment]
Wait a moment... we already have a markdownlint check (https://github.com/hamstar/Wikimate/pull/138, #146). Somehow it [failed](https://github.com/hamstar/Wikimate/actions/runs/5613291205/job/15208992761) to detect the issue. Could it be that the `**/*.md` glob is failing to detect the files at the root of the repo? Either way, we should look into making the output of that script verbose, so that we at least know which files it looked into (even if it found no issues).

**Comment by waldyrious** [comment]
Opened #154 to address the above.
